src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1057,15 +1055,15 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
 
           if (context.implicitsEnabled && !pt.isError && !tree.isErrorTyped) {
             // (14); the condition prevents chains of views
-            debuglog("inferring view from " + tree.tpe + " to " + pt)
-            inferView(tree, tree.tpe, pt, reportAmbiguous = true) match {
-              case EmptyTree =>
+            inferView(tree, tree.tpe, pt) match {
+              case EmptyTree => // didn't find a view -- fall through
               case coercion  =>
-                def msg = "inferred view from " + tree.tpe + " to " + pt + " = " + coercion + ":" + coercion.tpe
-                if (settings.logImplicitConv)
-                  context.echo(tree.pos, msg)
+                if (settings.debug || settings.logImplicitConv) {
+                  val msg = s"inferred view from ${tree.tpe} to $pt via $coercion: ${coercion.tpe}"
```

Comment:
I think this will create a throwaway string under `-Ydebug -XlogImplicitConv` (i.e. when `-Ylog:typer` is not enabled)